[APP] Tapp-103: CP initial implementation

michellemtchai opened the pull request:
**Please merge this after PR #105**

Please see Issue #103 for details.

chrislessard commented on api/db/migrate/20190516205625_add_columns_to_assignment.rb:
```diff
@@ -0,0 +1,10 @@
+  def change
+    add_column :assignments, :hours, :integer
+    add_column :assignments, :pay1, :float
+    add_column :assignments, :pay2, :float
```
It's unclear to me what is meant by "pay1" and "pay2". Can this be clarified with a comment or change of names ?

michellemtchai replied:
Actually, I plan to remove these attributes. Even though it was discussed during the meeting on Thursday that we should use them, there's apparently a plan to move them to a new table called `fund_source`. I'm kind of waiting for a final decision in case there's anymore changes...

chrislessard replied:
I see. Is there any harm in removing them now since they'll be find-able under version control anyways ? I would be tempted to remove them for now and keep a note somewhere when fund_source gets created. We can always put them back.

chrislessard commented on api/db/migrate/20190516205813_add_columns_to_offer.rb:
```diff
@@ -0,0 +1,10 @@
+  def change
+    add_column :offers, :hours, :integer
+    add_column :offers, :pay1, :float
+    add_column :offers, :pay2, :float
```
Same as above.

chrislessard commented on api/db/schema.rb:
```diff
@@ -36,6 +36,21 @@
+    t.datetime "updated_at", null: false
+    t.integer "hours"
+    t.float "pay1"
+    t.float "pay2"
```
Same as above.

chrislessard commented on api/db/schema.rb:
```diff
@@ -53,6 +68,23 @@
+    t.datetime "updated_at", null: false
+    t.integer "hours"
+    t.float "pay1"
+    t.float "pay2"
```
Same as above.

chrislessard commented on api/app/controllers/api/v1/assignments_controller.rb:
```diff
@@ -0,0 +1,91 @@
+
+module Api::V1
+  # Controller for Assignments
+	class AssignmentsController < ApplicationController
```
Please fix this formatting.

michellemtchai replied:
Do you mean formatting as in the tabbing?

chrislessard replied:
Yes !

chrislessard commented on api/app/assets/javascripts/api/v1/assignments.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Why are we generating .coffee files ? Are there plans to use this ? I would prefer these to be dropped these if not.

michellemtchai replied:
These were generated when I made the models in rails. Do you have a list of files that should be removed if I generate model through the rails command? Or is it preferred that I make these files manually?

michellemtchai replied:
I've removed the non-controller files. Does it look okay now?

chrislessard replied:
This looks good to me !

chrislessard reviewed (changes requested):
Hi Michelle, 

Thanks for all this great work ! I do have a few other nits : 

1. Please follow the commit format mentioned in the project README. Yours aren't far off, but we try to indicate both the nature of the commit and the affected region in the header, such as in "fix(docker): add PGDATA to .env" to make the commit long more consistent and clear. 

2. This is creeping towards the upper bound for PR size, as it becomes harder to review everything thoroughly in a reasonable time frame. IMO it would have been ok to keep the DB stuff and the route/code/test boilerplate separate, especially since DB work tends to be more contested.

michellemtchai committed:
feat(db): add assignments & offers

create assignment and offer table in accordance to description in issue #103

michellemtchai committed:
feat(api): create assignment & offer model

generate models for both assignment and offer in rails

michellemtchai committed:
fix(db): fix assignment & offer FK

remove the typo of 's' on the foreign key declaration in the migration file for assignment and offer table creation

michellemtchai committed:
feat(db): create unique index in assignment

make the applicant_id, position_id pair unique in the assignment table

michellemtchai committed:
fix(db): remove applicant & position in offer

remove applicant_id and position_id FK from offers table due to redundancy

michellemtchai committed:
fix(api): remove typo in assignment, offer model

remove 's' typo in belongs_to of assignment and offer model

michellemtchai committed:
feat(api): create route for assignment, offer

GET /api/v1/assignments

GET /api/v1/assignments/:id

POST /api/v1/assignments

PUT /api/v1/assignments/:id

GET /api/v1/offers

GET /api/v1/offers/:id

michellemtchai commented:
I've rewritten the commit messages. Does this look okay now?

michellemtchai committed:
fix(api): fix tabbing on assignment & offer controllers

fix inconsistent tabbing on assignment and offer controller

michellemtchai committed:
feat(db): create funding_sources
create funding_sources table based on Slack discussion

chrislessard commented:
@michellemtchai 

Hi Michelle, 

Sorry for the delay. This all looks great to me 🙂Thanks for all your hard work !
